chromadb/ingest/__init__.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+import pulsar.schema as schema
+
+
+class AddEmbedding(schema.Record):
+    _avro_namespace_ = 'chromadb.ingest'
+    id = schema.Union([schema.String(), schema.UUID(), schema.Bytes(16)])
+    embedding = schema.Array(schema.Float())
+    metadata = schema.Map(schema.String(), schema.String())
+    object_reference = schema.String()
+    object_inline = schema.Bytes()
+    update = schema.Boolean()
+
+
+class DeleteEmbedding(schema.Record):
+    _avro_namespace_ = 'chromadb.ingest'
+    id = schema.Union([schema.String(), schema.UUID(), schema.Bytes(16)])
+
+
+class Message(schema.Record):
+    _avro_namespace_ = 'chromadb.ingest'
+    messages = Schema.Array(schema.Union([AddEmbedding, DeleteEmbedding]))
```

Comment:
Does this mean that a message can intermingle read and write?

Reply:
A given message will either be an add or delete. Each topic can only have one top-level schema, so assuming we want reads and writes on the same topic (I think we need to), that schema needs to accommodate both kinds of messages.

Reply:
Cool. Just confirming, since technically this type would allow intermingling.